src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -5269,6 +5336,44 @@ void SetAttribute(XDocument doc, string path, string name, string value)
             }
         }
 
+        private static async Task<(long nrPeriods, long nrThrottled, long throttledTime)> GetCpuStats(string controller)
+        {
+            var result = await ProcessUtil.RunAsync("cat", $"/sys/fs/cgroup/cpu/{controller}/cpu.stat", throwOnError: false, captureOutput: true);
+
+            // nr_periods 3
+            // nr_throttled 3
+            // throttled_time 258313264
+
+            long nrPeriods = 0, nrThrottled = 0, throttledTime = 0;
+
+            using (var sr = new StringReader(result.StandardOutput))
+            {
+                var line = sr.ReadLine();
+
+                while (line != null)
+                {
+                    if (line.StartsWith("nr_periods"))
+                    {
+                        long.TryParse(line.Split(' ', 2, StringSplitOptions.RemoveEmptyEntries).Last(), out nrPeriods);
```

Comment:
defensive parsing?

Reply:
`TryParse`, `2` and `Last` just in case of unexpected valuesm then it would be `0` which is better than something I don't know can happen.

Did you want to suggest something or just wondering?